fluentd/event-exporter/watchers/consumer.go

```diff
@@ -0,0 +1,24 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watchers
+
+// WatchConsumer interface provides a way to act upon signals from watcher.
+type WatchConsumer interface {
```

Comment:
Instead of introducing this interface, please reuse "ResourceEventHandler" interface from from client-go/tools/cache.